xmtp/src/account.rs

```diff
@@ -1,19 +1,68 @@
+use std::vec;
+
 use serde::{Deserialize, Serialize};
-use vodozemac::olm::{Account, AccountPickle};
+use vodozemac::olm::{Account as OlmAccount, AccountPickle as OlmAccountPickle};
+
+use crate::Signable;
+
+type Address = String;
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub enum AssociationFormat {
+    Eip191,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub enum AssociationType {
+    Static,
```

Comment:
What are some other `AssociationType`s you have in mind. I don't understand the distinction between `AccountType` and `AccountFormat`

Reply:
In V2 the text used in link wallets and Identity keys was static. To support 1) a wider range of Associations and 2) versioning of the signature text its helpful to capture how the signature text ought to be validated. Expected values include:

- XmtpStaticV1
- SIWE 
- CAIP-122

**AssociationFormat** - The set of cryptographic standards to generate the Signature( EIP191, ED25519_BLAKE3, BIP340, etc)
**AssociationType** - The application level standards used to generate/validate the text which was signed.

*(The names could definitely use some work. Feedback welcome)*